pkg/controller/storagecluster/update.go

```diff
@@ -466,7 +466,12 @@ func (c *Controller) getUnavailableNumbers(
 			// for the purposes of update we ensure that the pod is
 			// both available and not terminating
 			if podutil.IsPodReady(pod) && pod.DeletionTimestamp == nil {
-				available = true
+				// Wait for MinReadySeconds after pod is ready.
+				if podutil.IsPodAvailable(pod, cluster.Spec.UpdateStrategy.RollingUpdate.MinReadySeconds, metav1.Now()) {
```

Comment:
we need to check the update strategy first, it can be `OnDelete` type and `RollingUpdate == nil`

Reply:
this is already inside of rolling update. For other updates we don't check unavailable pods.